### ISSUE TITLE
Parsing JSON payload with cppkafka

### ISSUE THREAD

**Issue by raghavendraramya** [mfontanini/cppkafka#276]
Hi, I am using cppkafka to have a Kafka consumer and connect to a Kafka server. The payload I receive is in a JSON format, so I am looking at converting it to a JSON object and then parse it. However, when I add in the code to create an object, I get the following error:

terminate called after throwing an instance of 'cppkafka::Exception'
  what():  Failed to create consumer handle: Failed to create thread: Success (0)
Aborted (core dumped)

If I comment the JSON object creation, the code runs fine and I can read messages from the Kafka server. I am confused as to why the JSON object creation has any relationship to the error.
The code I am using is given below. Lines 20 and 21, when commented, makes the code run fine and the consumer code is able to connect to a Kafka server and read the messages. When I uncomment the lines, I get the error above. 

I am thankful to any help in advance!
---------------------------------------------------------------------------------------------------------------------------

#include <stdexcept>
#include <iostream>
#include <csignal>
#include "json.hpp"
#include "cppkafka/consumer.h"
#include "cppkafka/configuration.h"

using std::string;
using std::exception;
using std::cout;
using std::endl;

using cppkafka::Consumer;
using cppkafka::Configuration;
using cppkafka::Message;
using cppkafka::TopicPartitionList;

void ParseKafkaDetection(std::string currentMessage)
{
	//json jsonObj;
	//std::stringstream currentMessage >> jsonObj;
	
	std::cout << "In ParseKafkaDetection" << std::endl;
	std::cout << currentMessage << std::endl;
}

int main() {

    string brokers;
    string topic_name = "my-topic";
    string group_id;

    Configuration config = {
        // The list of brokers we'll use
	{ "metadata.broker.list", "localhost" },

        // Our consumer's group
        { "group.id", "kafka-consumer-test" },

        // Disable auto commit
        { "enable.auto.commit", false },

        // Client group session timeout
        { "session.timeout.ms", 60000 }
    };

    // Create our consumer
    Consumer consumer(config);

   // Subscribe to the topic we want
   consumer.subscribe({ topic_name });

   // Poll forever
   while (true) {
   	Message msg = consumer.poll();
    	// Poll can return nothing, so make sure to check we got something
    	if (!msg) {
        	continue;
    	}
    	// There's an error
    	if (msg.get_error()) {
        	// rdkafka indicates EOF (end of partition) as errors,
        	// which doesn't really mean something went wrong
        	if (!msg.is_eof()) {
            	// This is an actual error, handle it properly
            		//handle_error(msg.get_error());
			printf("Error\n");
        	}
        	// Regardless, we need to keep going
        	continue;
    	}
    	// We actually received a message, process it
    	//cout << "Received message on partition " << msg.get_topic() << "/"
        // << msg.get_partition() << ", offset " << msg.get_offset() << endl;
	//cout << "Successfully produced message with payload " << msg.get_payload() << endl;
	ParseKafkaDetection(std::string(msg.get_payload()));	
		

  }

}

**Comment by jlcordeiro** [comment]
tested it with a random json parser I found on github and it works fine.
I bet the problem is in your json.hpp file (which you didn't attach). I'd look there.

If you need further help just drop that file in here also and I can have a look.

**Comment by raghavendraramya** [comment]
Thank you for getting back to me on this. Which parser did you use? Could you kindly send me a link?
[json_hpp.txt](https://github.com/mfontanini/cppkafka/files/6437519/json_hpp.txt)

I am using the nlohmann JSON parser. Please see the attached json.hpp.

**Comment by jlcordeiro** [comment]
yeah you're using the library wrong. I'm even surprised it is compiling for you, so I'd suggest you look at your build process and see if something funky is going on.

Either way please see below a couple of changes that made your code work for me, and printing out the received events, in json.

`diff bin/Main.txt.txt bin/main.cpp 
20,21c20,21
< 	//json jsonObj;
< 	//std::stringstream currentMessage >> jsonObj;
---
> 	nlohmann::json jsonObj = nlohmann::json::parse(currentMessage);
> 	std::cout << jsonObj.dump(4) << std::endl;



And here's a full example where I just removed unnecessary lines for brevity sake:

`#include <stdexcept>
#include <iostream>
#include <csignal>
#include "json.hpp"
#include "cppkafka/consumer.h"
#include "cppkafka/configuration.h"

using std::string;
using std::exception;
using std::cout;
using std::endl;

using cppkafka::Consumer;
using cppkafka::Configuration;
using cppkafka::Message;
using cppkafka::TopicPartitionList;

void ParseKafkaDetection(std::string currentMessage)
{
    nlohmann::json jsonObj = nlohmann::json::parse(currentMessage);
    std::cout << jsonObj.dump(4) << std::endl;
}

int main() {

    string brokers;
    string topic_name = "my-topic";
    string group_id;

    Configuration config = {
        { "metadata.broker.list", "localhost" },
        { "group.id", "kafka-consumer-test" },
        { "enable.auto.commit", false },
        { "session.timeout.ms", 60000 }
    };

    Consumer consumer(config);
    consumer.subscribe({ topic_name });

    while (true) {
        Message msg = consumer.poll();
        if (!msg) {
            continue;
        }
        // There's an error
        if (msg.get_error()) {
            if (!msg.is_eof()) {
                printf("Error\n");
            }
            continue;
        }
        ParseKafkaDetection(std::string(msg.get_payload()));	
    }
}
`
